dashboard/lib/services/lti.rb

```diff
@@ -192,6 +192,16 @@ def self.sync_section_roster(lti_integration, lti_section, nrps_section)
           nrps_member: nrps_member
         )
         user_was_new = user.new_record?
+        # Update name if different from the NRPS response
```

Comment:
Just so I understand correctly.
During roster sync, if a user is not a new record, your changes will always try assigning `user.name` for teacher, and `user.name` and `user.family_name` for students. 
Technically, we are not checking if there's a change in the LMS side, we just always assign name and family name, and then check if there are any changes in the user with the `had_changes ||= (user_was_new || user.changed?)`
Does that sound correct?